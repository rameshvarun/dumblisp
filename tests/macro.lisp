;; Macro with no arguments
(defmacro greet () (list 'print "Hello World!"))
(greet)
(greet)

;; Increment Macro
(defmacro incr (x)
  (list 'set x (list '+ x 1)))

(let ((i 0))
	(print i)
	(incr i)
	(incr i)
	(incr i)
	(print i))