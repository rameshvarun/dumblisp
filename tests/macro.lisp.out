Hello World!
Hello World!
0
3
